packages/block-library/src/query-total/edit.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	useBlockProps,
+	InspectorControls,
+	BlockControls,
+} from '@wordpress/block-editor';
+import {
+	PanelBody,
+	ToolbarGroup,
+	ToolbarDropdownMenu,
+} from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { resultsFound, displayingResults } from './icons';
+
+export default function QueryTotalEdit( { attributes, setAttributes } ) {
+	const { displayType } = attributes;
+
+	// Mock values.
+	const totalResults = 12;
+	const currentPage = 1;
+	const resultsPerPage = 10;
+
+	// Helper to calculate the range for "Displaying X – Y of Z".
+	const calculateRange = () => {
```

Comment:
This can be calculated once for the props used in the function and memoized with `useMemo` React hook.

Reply:
@gziolo I am thinking of getting rid of the function and completely hard coding the values for editor like this.

```js
if ( displayType === 'range-display' ) {
  return (
    <div>
      { __( 'Displaying 1 - 10 of 12' ) }
    </div>
  );
}
```

What are your views?

Reply:
To simplify everything I have removed this function & hard coded the output on the editor side

Reply:
Make sense 👍🏻